### PR TITLE
Update the Kubernetes version requirements.

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: docs/install/Knative-with-any-k8s.md [hunk 1]
```diff
@@ -10,11 +10,9 @@ using pre-built images.
 
 ## Before you begin
 
-Knative requires a Kubernetes cluster v1.11 or newer with the
-[MutatingAdmissionWebhook admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-controller)
-enabled. `kubectl` v1.10 is also required. This guide assumes that you've
-already created a Kubernetes cluster which you're comfortable installing _alpha_
-software on.
+Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.10 is also
```
Comment (samodell) [inline docs/install/Knative-with-any-k8s.md]: @mattmoor  Should we also bump up the required kubectl version?

Comment (mattmoor) [reply docs/install/Knative-with-any-k8s.md]: I don't know what informs our requirements there, so 🤷‍♂

**Comment by mattmoor** [comment]
I fixed up a few more places with 1.11 references.

File: docs/install/Knative-with-AKS.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ recommended configuration for a cluster is:
    az aks create --resource-group $RESOURCE_GROUP \
    --name $CLUSTER_NAME \
    --generate-ssh-keys \
-   --kubernetes-version 1.11.5 \
+   --kubernetes-version 1.14.0 \
```
Comment (dgerd) [inline docs/install/Knative-with-AKS.md]: Does this exist?

File: docs/install/Knative-with-AKS.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ You can find [guides for other platforms here](./README.md).
 
 ## Before you begin
 
-Knative requires a Kubernetes cluster v1.11 or newer. `kubectl` v1.10 is also
+Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.10 is also
```
Comment (dgerd) [inline docs/install/Knative-with-AKS.md]: From this kubectl docs:
```
You must use a kubectl version that is within one minor version difference of your cluster. For example, a v1.2 client should work with v1.1, v1.2, and v1.3 master. Using the latest version of kubectl helps avoid unforeseen issues.
```

Looks like this should be v1.13

File: docs/install/Knative-with-GKE.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ You can find [guides for other platforms here](./README.md).
 > [Cloud Run on GKE](https://cloud.google.com/run/docs/gke/setup) is a hosted
 > offering on top of GKE that builds around Istio and Knative Serving.
 
-Knative requires a Kubernetes cluster v1.11 or newer. `kubectl` v1.10 is also
+Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.10 is also
```
Comment (dgerd) [inline docs/install/Knative-with-GKE.md]:
```suggestion
Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.13 is also
```

File: docs/install/Knative-with-Gloo.md [hunk 5]
```diff
@@ -14,11 +14,9 @@ Gloo provides a complete gateway replacement for Istio and supports the full Kna
 
 ## Before you begin
 
-Knative requires a Kubernetes cluster v1.11 or newer with the
-[MutatingAdmissionWebhook admission controller](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-controller)
-enabled. `kubectl` v1.10 is also required. This guide assumes that you've
-already created a Kubernetes cluster which you're comfortable installing _alpha_
-software on.
+Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.10 is also
```
Comment (dgerd) [inline docs/install/Knative-with-Gloo.md]:
```suggestion
Knative requires a Kubernetes cluster v1.14 or newer. `kubectl` v1.13 is also
```

File: docs/install/Knative-with-AKS.md (hunk 3 above)
Comment (samodell) [reply docs/install/Knative-with-AKS.md]: Hm, if it's saying that v1.2 works with v1.1 and v1.3, then somebody using kubectl v1.1 is still good to go with k8s v1.1x, no matter what `x` is, right?

Comment (mattmoor) [reply docs/install/Knative-with-AKS.md]: I think Dan's right, should we just remove this wording though?

**Comment by mattmoor** [comment]
This is RFAL

**Pull request merged**